xblock_skytap/public/js/src/skytap.js

```diff
@@ -34,13 +34,29 @@ function SkytapXBlock(runtime, element) {
         launchXHR = $.post(handlerUrl, JSON.stringify(keyboardLayout))
             .success(function(response) {
                 var url = response.sharing_portal_url;
-                var sharingPortal = window.open(url, '_blank');
-                if (sharingPortal === null) {
-                    alert(
-                        gettext("The browser's popup blocker prevented the exercise environment from being launched.")
-                    );
+
+                /*
+                 The standard behaviour is to open the exercise environment in a new tab using a popup,
+                 to allow the user to keep the course tab open too. However on iOS devices, popups are
+                 blocked by default and the user is not informed that the popup failed to open.
+                 Therefore for iOS devices a redirect is used instead.
+                 */
+                var iOS = navigator.userAgent.match(/(iPod|iPhone|iPad)/);
```

Comment:
Can we also add checks for Android and Windows phone user agents if there are good identifiers we can use?